test/test_transform_ops.py

```diff
@@ -0,0 +1,265 @@
+        decoder = create_from_file(str(NASA_VIDEO.path))
+        with pytest.raises(
+            RuntimeError,
+            match="Transforms are only supported for CPU devices.",
+        ):
+            add_video_stream(decoder, device="cuda", transform_specs="resize, 100, 100")
+
+    def test_transform_fails(self):
+        decoder = create_from_file(str(NASA_VIDEO.path))
+        with pytest.raises(
+            RuntimeError,
+            match="Invalid transform spec",
+        ):
+            add_video_stream(decoder, transform_specs=";")
+
+        with pytest.raises(
+            RuntimeError,
+            match="Invalid transform name",
+        ):
+            add_video_stream(decoder, transform_specs="invalid, 1, 2")
+
+    def test_resize_transform_fails(self):
+        decoder = create_from_file(str(NASA_VIDEO.path))
+        with pytest.raises(
+            RuntimeError,
+            match="must have 3 elements",
+        ):
+            add_video_stream(decoder, transform_specs="resize, 100, 100, 100")
+
+        with pytest.raises(
+            RuntimeError,
+            match="must be a positive integer",
+        ):
+            add_video_stream(decoder, transform_specs="resize, -10, 100")
+
+        with pytest.raises(
+            RuntimeError,
+            match="must be a positive integer",
+        ):
+            add_video_stream(decoder, transform_specs="resize, 100, 0")
+
+        with pytest.raises(
+            RuntimeError,
+            match="cannot be converted to an int",
+        ):
+            add_video_stream(decoder, transform_specs="resize, blah, 100")
+
+        with pytest.raises(
+            RuntimeError,
+            match="out of range",
+        ):
+            add_video_stream(decoder, transform_specs="resize, 100, 1000000000000")
+
+    def test_crop_transform(self):
+        # w=300, h=200, x=50, y=35
+        crop_spec = "crop, 300, 200, 50, 35"
+        crop_filtergraph = "crop=300:200:50:35:exact=1"
+
+        decoder_crop = create_from_file(str(NASA_VIDEO.path))
+        add_video_stream(decoder_crop, transform_specs=crop_spec)
+
+        decoder_full = create_from_file(str(NASA_VIDEO.path))
+        add_video_stream(decoder_full)
+
+        for frame_index in [0, 15, 200, 389]:
+            frame, *_ = get_frame_at_index(decoder_crop, frame_index=frame_index)
+            frame_ref = NASA_VIDEO.get_frame_data_by_index(
+                frame_index, filters=crop_filtergraph
+            )
+
+            frame_full, *_ = get_frame_at_index(decoder_full, frame_index=frame_index)
+            frame_tv = v2.functional.crop(
+                frame_full, top=35, left=50, height=200, width=300
+            )
+
+            assert frame.shape == (3, 200, 300)
+            assert frame_ref.shape == (3, 200, 300)
+            assert frame_tv.shape == (3, 200, 300)
+
+            assert_frames_equal(frame, frame_tv)
+            assert_frames_equal(frame, frame_ref)
```

Comment:
On the test above: I trust that you hardcoded the H and W values properly, but it might be a bit safer to set those as variables, like

```py
        W, H, X, Y = 300, 200, 50, 35
        crop_spec = f"crop, {W}, {H}, {X}, {Y}"
        crop_filtergraph = f"crop={W}:{h}:{X}:{Y}:exact=1"

....

assert frame.shape == (3, H, W)
```